### ISSUE TITLE
Class_weight mathematical formula?

### ISSUE THREAD

**Issue by 4wa** [keras-team/keras#12736]
I'm using class_weights to minimize the problem of imbalanced dataset with multi-class of 10 classes.
However, there is no much difference to the results when applying vs not applying the class_weighted feature during "fit". 
I would like to know what is the mathematical formula used to obtain the class_weights on the loss function "categorical_crossentropy"?

The only explanation I could find in Keras documentation is the following statement:

> class_weight: Optional dictionary mapping class indices (integers) to a weight (float) value, used for weighting the loss function (during training only). This can be useful to tell the model to "pay more attention" to samples from an under-represented class.

There is really nothing related to how its calculated during the loss function!
Also the way to calculate the class weight "from my own understanding" is to apply simple division:
total num of instances / number of instances in one class = the value to feed the class_weight for this particular class.
I'm using integers class values for example: 
class_weights = { 0 = 4.5,
                            1= 35.4,
                            2= 34.}
While my y labels are actually converted to  np_utils.to_categorical(train_target, 10). 

So, am I doing the whole process in a wrong way? 
If so which part?!

**Comment by saurabheights** [comment]
Dear, you can check this https://github.com/angeladai/3DMV/blob/master/3dmv/train.py#L89. Basically, Weight = 1/(1 + log(count)) 
Log prevents too much weightage to the undersampled class.
I have also seen people use simple counting(no log) as weights scheme as u mentioned.

> However, there is no much difference to the results when applying vs not applying the class_weighted feature during "fit".

Do you see any affect in time it takes to reach the result?

**Comment by 4wa** [comment]
> Dear, you can check this https://github.com/angeladai/3DMV/blob/master/3dmv/train.py#L89. Basically, Weight = 1/(1 + log(count))
> Log prevents too much weightage to the undersampled class.
> I have also seen people use simple counting(no log) as weights scheme as u mentioned.
> 
> > However, there is no much difference to the results when applying vs not applying the class_weighted feature during "fit".
> 
> Do you see any affect in time it takes to reach the result?

Not so much!
However I found this post in here: https://github.com/keras-team/keras/issues/3653
Do you think the fact that I'm classifying images with 3D has any relation to the not so much improving performance?!
